virtool/users/data.py

```diff
@@ -161,39 +171,103 @@ async def create(
         :param handle: the requested handle for the user
         :param password: a password
         :param force_reset: force the user to reset password on next login
+        :param  b2c_user_attributes: Azure b2c user attributes used to describe a user
         :return: the user document
         """
-        document = await create_user(self._mongo, handle, password, force_reset)
+        async with both_transactions(self._mongo, self._pg) as (
+            mongo_session,
+            pg_session,
+        ):
+            now = virtool.utils.timestamp()
+            display_name = ""
+            given_name = ""
+            family_name = ""
+            oid = ""
+
+            document = await create_user(
+                self._mongo,
+                handle,
+                password,
+                force_reset,
+                b2c_user_attributes=b2c_user_attributes,
+                session=mongo_session,
+            )
+
+            if b2c_user_attributes:
+                display_name = b2c_user_attributes.display_name
+                given_name = b2c_user_attributes.given_name
+                family_name = b2c_user_attributes.family_name
+                oid = b2c_user_attributes.oid
+
+            pg_session.add(
+                SQLUser(
+                    legacy_id=document["_id"],
+                    handle=handle,
+                    password=document["password"] if password else None,
+                    force_reset=force_reset,
+                    last_password_change=now,
+                    b2c_display_name=display_name,
+                    b2c_given_name=given_name,
+                    b2c_family_name=family_name,
+                    b2c_oid=oid,
+                )
+            )
+
         return await self.get(document["_id"])
 
+    async def delete(self, legacy_id: str):
+        """
+        Delete the user with the matching legacy_id.
+
+        :param legacy_id: legacy id of the user to delete
+        :return:
+        """
+        async with both_transactions(self._mongo, self._pg) as (
+            mongo_session,
+            pg_session,
+        ):
+            await pg_session.execute(
+                delete(SQLUser).where(SQLUser.legacy_id == legacy_id)
+            )
+            await self._mongo.users.delete_one(
+                {"_id": legacy_id}, session=mongo_session
+            )
```

Comment:
Ian wanted me to "Update the user create, update, and delete methods, as well as any other methods that affect users.  For now, we will not be modifying how user data is read." but there wasn't a delete before, so I added it. But I have now removed it again because I see the issues with it.